sourcegraph/sourcegraph
# [Part 2 for #issues/27922] Add browser extension support for cloud private repositories

Closes https://github.com/sourcegraph/sourcegraph/issues/27922. Blocked by https://github.com/sourcegraph/sourcegraph/pull/28387.

### Description
This PR:
- Adds back (secure) support for private repositories on Sourcegraph Cloud
> Note: as a **positive** side-effects:
> - browser extension does not override GitHub native tooltips when the repository is not cloned on Cloud
> - browser extension is immediately reinitialized on all open/active pages when Sourcegraph URL changes

### How to test
- `sg run bext`
- Set the extension to cloud URL (https://sourcegraph.com)
  - Open any public repo and make sure the browser extension works properly
  - Open any private repo which is cloned in Cloud and make sure the browser extension works properly
  - Open any private repo which is NOT cloned in Cloud and **make sure the browser extension does NOT work and does not make any requests except `SecureCheckRepoCloned` in extension background script -> Network tab**
- Set the extension to a self-hosted URL (https://k8s.sgdev.org)
  - Make that browser extension works as previously

### Screenshots

| Scenario | Screenshot |
| --: | :-- |
| cloud URL & private repo is cloned | ![image](https://user-images.githubusercontent.com/6717049/145415064-004f3abf-15d3-456b-954f-2863b8a1c309.png) |
| cloud URL & private repo is NOT cloned | ![image](https://user-images.githubusercontent.com/6717049/145415442-79a03e8b-de72-4586-8197-e26a56ea0970.png) |
| cloud URL & private repo && not sign in | ![image](https://user-images.githubusercontent.com/6717049/145416418-6a2eaa46-b7b3-433d-b464-21c02841c4ec.png) |
| cloud URL && public repo | ![image](https://user-images.githubusercontent.com/6717049/145414837-1f4931b0-507d-4c2a-8a43-dbeaf3dff5bf.png) |
| self-hosted URL | everything remains the same as previously |
